src/qbe_ir_generator.cpp

```diff
@@ -822,14 +822,49 @@ void QbeIrGenerator::Visit(const BinaryExprNode& bin_expr) {
   int left_num = num_recorder.NumOfPrevExpr();
   bin_expr.rhs->Accept(*this);
   int right_num = num_recorder.NumOfPrevExpr();
-  int num = NextLocalNum();
-  // TODO: use the correct instruction for specific data type:
-  // 1. signed or unsigned: currently only supports signed integers.
-  // 2. QBE base data type 'w' | 'l' | 's' | 'd': currently only supports 'w'.
-  WriteInstr_("{} =w {} {}, {}", FuncScopeTemp{num},
-              GetBinaryOperator(bin_expr.op), FuncScopeTemp{left_num},
-              FuncScopeTemp{right_num});
-  num_recorder.Record(num);
+  // The && operator shall yield 1 if both of its operands compare unequal to
+  // 0; otherwise, it yields 0; The || operator shall yield 1 if either of its
+  // operands compare unequal to 0; otherwise, it yields 0.
+  if (bin_expr.op == BinaryOperator::kLand ||
+      bin_expr.op == BinaryOperator::kLor) {
```

Comment:
How about placing an explanation of why we need this extra condition above the condition itself and moving the implementation details inside the loop?

```suggestion
  // Due to the lack of direct support for logical operators in QBE, we
  // implement logical expressions using comparison and jump instructions.
  if (bin_expr.op == BinaryOperator::kLand ||
      bin_expr.op == BinaryOperator::kLor) {
     // The && operator yields 1 if both of its operands compare unequal to 0;
     // otherwise, it yields 0. The || operator yields 1 if either of its
     // operands compare unequal to 0; otherwise, it yields 0.
```